pkg/controller/node/nodecontroller.go

```diff
@@ -141,6 +141,8 @@ type NodeController struct {
 	daemonSetStore listers.StoreToDaemonSetLister
 	// allocate/recycle CIDRs for node if allocateNodeCIDRs == true
 	cidrAllocator CIDRAllocator
+	// manages taints
+	taintManager *TaintController
```

Comment:
Please add a detailed description (from our emails) of why the TaintController is special cased to the NodeController, as this is different then all other controllers. 

/cc @aveshagarwal